buildchain/buildchain/docker_command.py

```diff
@@ -77,6 +77,24 @@ def decorated_task(*args: Any, **kwargs: Any) -> Optional[TaskError]:
     return wrapped_task
 
 
+# The call method is not counted as a public method
+# pylint: disable=too-few-public-methods
+class DockerExists:
+    """A class providing an image existence check through the API client."""
+    def __init__(self, name: str, version: str):
```

Comment:
I would take either a single parameter `tag` (or whatever is the proper naming) or directly an `ContainerImage` instance.